### PR TITLE
ADAPT-3167: Use TW form plugin for checkbox styling

### PR DESCRIPTION
**Pull request by yvonnetangsu**
# READY FOR REVIEW

# Summary
- Remove `su-custom-checkbox` style which is a custom TW component. Also remove the checkbox svg image that is in the static/ folder
- Install official TW form plugin so checkboxes can be styled more easily with TW classes
- Clean up checkboxes styling; fixup search input boxes style so they are the same as before

# Review By (Date)
- When does this need to be reviewed by?

# Criticality
- How critical is this PR on a 1-10 scale? Also see [Severity Assessment](https://stanfordits.atlassian.net/browse/D8CORE-1705).
- E.g., it affects one site, or every site and product?

# Review Tasks

## Setup tasks and/or behavior to test

1. Go to https://deploy-preview-134--stanford-alumni.netlify.app/search/
2. Click on the checkboxes on the left sidebar and they should look like this:
Screen Shot 2021-08-16 at 6 12 34 PM

3. Tab from one checkbox to the next and they should have a focus ring around

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: src/components/search/searchFacet.js [hunk 1]
```diff
@@ -67,7 +62,7 @@ const SearchFacet = ({
             value={option.name}
             name={attribute}
             defaultChecked={selectedOptions.includes(option.name)}
-            className="su-checkbox su-mr-15 su-max-w-[1.7rem] su-cursor-pointer su-custom-checkbox su-max-h-[1.7rem] su-appearance-none !su-border su-rounded-[0.3rem]"
+            className="su-form-checkbox su-text-digital-red-light su-mr-10 su-w-[1.5rem] su-h-[1.5rem] su-cursor-pointer su-rounded su-border-black-40 focus:su-ring-0 focus:su-ring-transparent focus-visible:su-ring-2 focus-visible:su-ring-digital-blue-light"
```
Comment (yvonnetangsu) [inline src/components/search/searchFacet.js:65]: `focus:` styles is for clicking with mouse (no focus ring), `focus-visible:` styles for tabbing from checkbox to checkbox; su-form-checkbox is from TW plugin, basically their "reset" styles for form elements so they can be styled with TW classes easily.

File: src/components/search/searchFieldModal.js [hunk 2]
```diff
@@ -17,8 +17,8 @@ const SearchFieldModal = React.forwardRef((props, ref) => {
 
   const wrapperClasses = `su-border-0 su-border-b-2 su-border-solid su-border-black-10`;
 
-  const inputClasses = `su-bg-transparent su-text-44 su-text-black-40 su-w-full su-flex-1
-  su-rs-px-1 su-py-10 su-text-m4 focus:su-outline-none`;
+  const inputClasses = `su-border-0 su-bg-transparent su-text-44 su-text-black-40 su-w-full su-flex-1
+  su-rs-px-1 su-py-10 su-text-m4 focus:su-outline-none focus:su-ring-0 focus:su-ring-transparent`;
```
Comment (yvonnetangsu) [inline src/components/search/searchFieldModal.js:21]: Clean up the search input text box border to remove some TW form default styles

File: src/components/page-types/searchPage.js [hunk 3]
```diff
@@ -216,7 +216,7 @@ const SearchPage = (props) => {
   const clearBtnClasses = `su-flex su-items-center su-bg-transparent hover:su-bg-transparent su-text-21 su-font-semibold
   hover:su-text-black su-border-none su-text-black-70 su-p-0 focus:su-bg-transparent focus:su-text-black-70 su-rs-mr-1`;
 
-  const inputClasses = `su-text-30 su-w-full su-flex-1 su-rs-px-1 su-py-10 su-text-m2 su-outline-none`;
+  const inputClasses = `su-border-0 su-text-30 su-w-full su-flex-1 su-rs-px-1 su-py-10 su-text-m2 su-outline-none focus:su-ring-0 focus:su-ring-transparent`;
```
Comment (yvonnetangsu) [inline src/components/page-types/searchPage.js:219]: Clean up the modal search input text box border to remove some TW form default styles

File: gatsby-ssr.js [hunk 4]
```diff
@@ -4,14 +4,12 @@
...
 // Contexts.
-import { GlobalStateProvider } from './src/contexts/GlobalContext';
+import { GlobalStateProvider } from "./src/contexts/GlobalContext";
 
 // Exports.
 export const wrapRootElement = ({ element }) => (
-  <GlobalStateProvider>
-    {element}
-  </GlobalStateProvider>
+  <GlobalStateProvider>{element}</GlobalStateProvider>
```
Comment (yvonnetangsu) [inline gatsby-ssr.js:14]: Just linting

File: src/components/search/searchFacet.js [hunk 5]
```diff
@@ -43,15 +43,10 @@ const SearchFacet = ({
...
-          "lg:su-pb-0 su-border-b lg:su-border-0 su-border-black-20"
+        "su-mb-[24px] lg:su-mb-[66px] su-pb-[28px] lg:su-pb-0 su-border-b lg:su-border-0 su-border-black-20"
       }`}
     >
-      <Heading
-        level={2}
-        weight="semibold"
-        className="su-text-18 lg:su-text-23 su-mb-18 lg:su-mb-21"
-      >
+      <Heading level={2} weight="semibold" className="su-rs-mb-0 su-text-m0">
```
Comment (yvonnetangsu) [inline src/components/search/searchFacet.js:49]: Just spacing/font size fixup

**Comment by yvonnetangsu** [comment]
@Lyafel Replacing some checkbox styling using official TW form plugin.

**Review by sherakama — APPROVED** [review]
GTG

**Comment by sherakama** [comment]
@yvonnetangsu This new form package would be a good package to add to the core decanter v7 branch.

**Comment by yvonnetangsu** [comment]
> @yvonnetangsu This new form package would be a good package to add to the core decanter v7 branch.

Actually I believe it's been installed on core v7 already, but I haven't had a chance to start working on the form elements back then 😄

**Pull request merged**